logic/service/blueprint-data-mapping.js

```diff
@@ -0,0 +1,140 @@
+var DataMapping = require("logic/service/data-mapping").DataMapping;
+
+/**
+ * Maps raw data to data objects of a specific type by using a blueprint.
+ **
+ * @class
+ * @extends external:Montage
+ */
+exports.BlueprintDataMapping = DataMapping.specialize(/** @lends BlueprintDataMapping.prototype */ {
+
+    _blueprint: {
+        value: undefined
+    },
+
+    /***************************************************************************
+     * Mapping
+     */
+
+    /**
+     * Convert raw data to data objects of an appropriate type.
+     *
+     * Subclasses should override this method to map properties of the raw data
+     * to data objects, as in the following:
+     *
+     *     mapRawDataToObject: {
+     *         value: function (object, data) {
+     *             object.firstName = data.GIVEN_NAME;
+     *             object.lastName = data.FAMILY_NAME;
+     *         }
+     *     }
+     *
+     * The default implementation of this method copies the properties defined
+     * by the raw data object to the data object.
+     *
+     * @method
+     * @argument {Object} object - An object whose properties must be set or
+     *                             modified to represent the raw data.
+     * @argument {Object} data   - An object whose properties' values hold
+     *                             the raw data.
+     * @argument {?} context     - A value that was passed in to the
+     *                             [addRawData()]{@link DataService#addRawData}
+     *                             call that invoked this method.
+     */
+    mapRawDataToObject: {
+        value: function (data, object, context) {
```

Comment:
Right now we are iterating over the blueprint's properties.  Should we instead iterate over the object descriptor's property descriptors?  Or should we only iterate over the mappings? 

If it is the former, I suppose we would skip the property descriptors that are relationships?  If it is the latter then the developer would need to create a mapping for each property they want to map.  Which doesn't seem to unreasonable (that's how it works in Contour).  I suppose two one way bindings is preferred so that you can define the number of mappings in each direction?